packages/block-library/src/block/edit.js

```diff
@@ -84,6 +97,56 @@ export default function ReusableBlockEdit( { attributes: { ref }, clientId } ) {
 
 	const blockProps = useBlockProps();
 
+	// local states for gudie modal
+	const [ isGudieOpen, setIsGudieOpen ] = useState( false );
+
+	// Get all the unsaved Records
+	const dirtyRecords = useSelect( ( select ) => {
+		return select( coreStore ).__experimentalGetDirtyEntityRecords();
```

Comment:
Since we're checking edits (dirty records) for a single entity, I think we can use the `hasEditsForEntityRecord` selector here.

Something like this:

```js
select( coreStore ).hasEditsForEntityRecord( 'postType', 'wp_block', ref );
```